mesosphere/kudo-cassandra-operator
# Implement Prometheus Metrics

## What changes are proposed in this PR? 
Resolves  [DCOS-59498](https://jira.mesosphere.com/browse/DCOS-59498) 

This PR Implement Prometheus Metrics for kudo Cassandra operator. we have used [criteo cassandra_exporter](https://github.com/criteo/cassandra_exporter) to export Cassandra metrics.

To implement this below decisions are made : 
1. created different container in same pod to run metrics, and added related parameters.
2. added config file cassandra-exporter-config-yml and downloaded jar for criteo cassandra exporter.
3. Added label kudo.dev/servicemonitor: "true" and created  service-monitor.yaml file to create Service Monitor object for service discovery.

## How were these changes tested?
Manually tested and Below are steps to check 
1. Spin up Konvoy cluster and ensure cluster config is applied for kubectl.
2. Install KUDO Cassandra operator
3. Run command to create service-monito: kubectl create -f resources/service-monitor.yaml
4. Wait 10-20 sec, and go to Prometheus dashboard. Type there Cassandra you will see "cassandra_stats".
